### PR TITLE
Switch to use Stepper component to add app from Git

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:
This PR makes a change on the form to add an app from Git to use the [Stepper API](https://v4.mui.com/components/steppers/).
This allows you to clearly separate the two actions that the you should take.

https://user-images.githubusercontent.com/19730728/146480370-51911562-5678-44b4-9895-3b3408c6454f.mov

t this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nakabonne** [comment]
I made a minor change to close the second stepper whenever you re-select Piped.
Kapture 2021-12-17 at 13 13 09

File: pkg/app/web/src/components/application-form/index.tsx [hunk 1]
```diff
@@ -609,6 +625,7 @@ const UnregisteredApplicationList: FC<ApplicationFormProps> = memo(
         setSelectedPipedId(options.pipedId);
         setSelectedKind(options.kind);
         setSelectedCloudProvider(options.cloudProvider);
+        options.cloudProvider ? setActiveStep(1) : setActiveStep(0);
```
Comment (khanhtc1202) [inline pkg/app/web/src/components/application-form/index.tsx:628]:
```suggestion
        setActiveStep(options.cloudProvider ? 1 : 0);
```
nits

**Comment by khanhtc1202** [comment]
Nice work 👏  LGTM with a small nit
/lgtm

File: pkg/app/web/src/components/application-form/index.tsx (hunk 1 above)
Comment (nakabonne) [reply pkg/app/web/src/components/application-form/index.tsx]: Cool!

**Comment by khanhtc1202** [comment]
🚀 
/lgtm

**Comment by nghialv** [comment]
Great. Looks beautiful.
/approve

**Pull request merged**